src/Microsoft.Extensions.Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -7,9 +7,42 @@
 
 namespace Microsoft.Extensions.Configuration
 {
-    public static class MemoryConfigurationExtensions
+    public static class ConfigurationBuilderExtensions
     {
         /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration)
+        {
+            if (configurationBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(configurationBuilder));
+            }
+            configurationBuilder.Add(new ChainedConfigurationProvider(configuration));
+            return configurationBuilder;
+        }
+
+        /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <param name="key">The key of the root of the configuartion to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration, string key)
```

Comment:
Do not confuse the `prefix` argument with a `key` to get a subsection. From what I remember the prefix was supposed to be pre-pended to the keys of the underlying IConfiguration to produce the keys of the wrapping configuration, e.g.:

``` C#
hostingConfiguration["X"] = "Y";
var appConfiguration = new ConfigurationBuilder().AddConfiguration("hosting_",hostingConfiguration).Build();
Assert.Equal(appConfiguration["hosting_X"], "X"); 
``` 

Unfortunately I don't see any notes on this in the request and I can't find any notes in the related PR https://github.com/aspnet/Hosting/pull/448, but hopefully @DamianEdwards, @davidfowl or @Tratcher can clarify.